lightning/backend/local.go

```diff
@@ -131,6 +140,60 @@ type local struct {
 	ingestConcurrency *worker.Pool
 	batchWriteKVPairs int
 	checkpointEnabled bool
+
+	tcpConcurrency int
+}
+
+// connPool is a lazy pool of gRPC channels.
+// When `Get` called, it lazily allocates new connection if connection not full.
+// If it's full, then it will return allocated channels round-robin.
+type connPool struct {
+	*sync.Mutex
```

Comment:
```suggestion
	mu sync.Mutex
```

no need to put the mutex in a pointer since the pool is always used as `*connPool`

also no need to expose the mutex methods to public.